### PR TITLE
Support warnings for secrets

### PR DESCRIPTION
**Pull request by pleshakov**
### Proposed changes
- Support warnings for TLS and JWK secrets in Ing
- Support warnings for TLS secrets in VS

### CODE REVIEW

**Review by mikestephen — APPROVED** [review]
Looks good to me!

File: internal/configs/configurator.go [hunk 1]
```diff
@@ -1188,80 +1201,92 @@ func generateApResourceFileContent(apResource *unstructured.Unstructured) []byte
...
 		if err != nil {
-			return fmt.Errorf("Error adding or updating mergeableIngress %v/%v: %v", m.Master.Ingress.Namespace, m.Master.Ingress.Name, err)
+			return allWarnings, fmt.Errorf("Error adding or updating mergeableIngress %v/%v: %v", m.Master.Ingress.Namespace, m.Master.Ingress.Name, err)
 		}
+		allWarnings.Add(warnings)
 	}
 
 	if err := cnf.nginxManager.Reload(nginx.ReloadForOtherUpdate); err != nil {
-		return fmt.Errorf("Error when reloading NGINX when updating %v: %v", resource.GetKind(), err)
+		return allWarnings, fmt.Errorf("Error when reloading NGINX when updating %v: %v", resource.GetKind(), err)
```
Comment (Dean-Coakley) [inline internal/configs/configurator.go:1224]: Would it be good to include resNs/resName here?
```suggestion
		return allWarnings, fmt.Errorf("Error when reloading NGINX when updating %v %v/%v: %v", resource.GetKind(), resource.GetNamespace(), resource.GetName(), err)
```

File: internal/configs/ingress.go [hunk 2]
```diff
@@ -246,16 +252,19 @@ func generateNginxCfg(ingEx *IngressEx, apResources map[string]string, isMinion
 			Annotations: ingEx.Ingress.Annotations,
 		},
 		SpiffeClientCerts: staticParams.NginxServiceMesh && !cfgParams.SpiffeServerCerts,
-	}
+	}, allWarnings
 }
 
-func generateJWTConfig(secretRefs map[string]*secrets.SecretReference, cfgParams *ConfigParams, redirectLocationName string) (*version1.JWTAuth, *version1.JWTRedirectLocation) {
+func generateJWTConfig(owener runtime.Object, secretRefs map[string]*secrets.SecretReference, cfgParams *ConfigParams,
```
Comment (Dean-Coakley) [inline internal/configs/ingress.go:258]:
```suggestion
func generateJWTConfig(owner runtime.Object, secretRefs map[string]*secrets.SecretReference, cfgParams *ConfigParams,
```

**Review by Dean-Coakley — APPROVED** [review]

File: internal/configs/ingress.go (hunk 2 above)
Comment (pleshakov) [reply internal/configs/ingress.go:258]: good catch

File: internal/configs/configurator.go (hunk 1 above)
Comment (pleshakov) [reply internal/configs/configurator.go:1224]: same enhancement can be applied to the other app protect related methods. better to do it in a separate PR

**Pull request merged**